test/fold.jl

```diff
@@ -0,0 +1,51 @@
+using Test
+using Distributions, Random
+import Distributions.fold_value
+import Distributions.unfold_value
+
+
+# Constructor works
+@test folded(Normal(1.0,2.0),1.0) isa Folded
+
+# Testing if the fold_value function and unfold_value functions fold the 
+# given value properly
+@testset "Folding and unfolding value in the domain" begin
+	@test fold_value(-10, folded(Normal(3,4),0)) == 10
+	@test fold_value( 5 , folded(Normal(3,4),0)) == 5
+	@test fold_value( 4, folded(Normal(2,3),10, keep_right=false)) == 4
+	@test fold_value( 16, folded(Normal(2,3),10, keep_right=false)) == 4
+	@test unfold_value(10, folded(Normal(3,4),0)) == -10
+	@test unfold_value( 5 , folded(Normal(3,4),0)) == -5
+	@test unfold_value( 16, folded(Normal(2,3),10, keep_right=false)) == 16
+	@test unfold_value( 4, folded(Normal(2,3),10, keep_right=false)) == 16
+end 
+
+# Testing if the insupport function works
+@testset "Folded insupport, minimum and maximum functions" begin
+	@test insupport(folded(Normal(2,1),0),-10) == false
+	# Check that the crease is always the minimum of a folded distrbution
+	@test all(((x->Distributions.minimum(folded(Normal(2,1),x))).(-10:0.1:10)) .≈ collect(-10:0.1:10))
+	# Check that the crease is always the maximum of a distrbution folded to the right
+	@test all(((x->Distributions.maximum(folded(Normal(2,1),x,keep_right=false))).(-10:0.1:10)) .≈ collect(-10:0.1:10))
+end
+
+@testset "Folded Rand functions" begin
```

Comment:
Same as for the above--we need tests that check if `rand` is working correctly, e.g. testing if a big sample from a folded normal distribution has the right distribution.